core/src/main/scala/org/apache/spark/sql/snappyParsers.scala

```diff
@@ -630,14 +633,21 @@ class SnappyParser(caseSensitive: Boolean)
  * Snappy dialect uses a much more optimized parser and adds SnappyParser
```

Comment:
You mean for SnappyContext? Thread local is nothing but heaps of trouble in such cases -- we have seen it over the years with Gem TXState and similar. This should be a general rule for us with very careful consideration if an exception has to be made. I can see a use in SnappyHiveCatalog though not sure if we really need that -- @kneeraj  can you tell the reason?

There is no restriction that user cannot have two SnappyContext's in the same flow of code, and then with ThreadLocal's we have to engage in a never-ending game of fixing the threadlocal from context and vice-versa (e.g. see GemFireTransaction.reattachTransaction, its call hierarchy from TransactionResourceImpl.setupContextStackAndReattach in snappy-store). Contexts in general are there to avoid this ugliness and we should stick by them. If need be, lets have explicit refs of context in analyser and other parts where required.